pulsar-client/src/main/java/org/apache/pulsar/client/impl/TypedMessageBuilderImpl.java

```diff
@@ -284,7 +284,7 @@ public MessageMetadata getMetadataBuilder() {
 
     public Message<T> getMessage() {
         beforeSend();
-        return MessageImpl.create(msgMetadata, content, schema, producer.topic);
+        return MessageImpl.create(msgMetadata, content, schema, producer != null ? producer.getTopic() : null);
```

Comment:
In what case can we get that producer is null? 

the field is defined as: 

```
    private transient final ProducerBase<?> producer;
```

and we create the message builder starting from the producer. Would that mean that the message builder was serialized and passed around?

Reply:
It happens when some legacy applications want to create a message using internal `TypedMessageBuilderImpl` API without initializing producer. eg: `pulsarMessageBuilder = new TypedMessageBuilderImpl(null, Schema.BYTES);`. we have few legacy API including wrapper on pulsar-storm which uses it to maintain backward compatibility as pulsar-v2 API has changed from v1 and we are not maintaining v1 API with all changes. so, it handles NPE when application tries to create message without initializing producer.